src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -7,89 +7,357 @@
 
 namespace System.Text.Json
 {
-    public partial class JsonNumber : JsonNode
+    public partial class JsonNumber : JsonNode, IEquatable<JsonNumber>
     {
+        private string _value;
+
         public JsonNumber() { }
```

Comment:
The default ctor here is the only time that `_value` can be `null`. Consider removing it, or initializing `_value` to `"0"`.